### PR TITLE
list functionality changes with profile in xsd

### PR DESCRIPTION
**Pull request by bernt-matthias**
Always found that docs on changes with profile versions is a bit to distributed. So here is a start. Not sure if its the best place.

### CODE REVIEW

**Comment by jmchilton** [comment]
This is great, thanks so much!

**Review by jmchilton — APPROVED** [review]

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 1]
```diff
@@ -181,14 +181,48 @@ set this to have values such as ``data_source`` or ``manage_data``.</xs:document
...
+- Introduce `provided_metadata_style` with default `"default"` before `"legacy"`.
+
+18.09 
+
+- References to other inputs need to be fully qualified by using `|`.
+- do not allow provided but illegal default values.
+- Do not use Galaxy python environment for `manage_data` tools.
+
+19.05
+- Use python 3.5 as default (before 2.7)
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/xsd/galaxy.xsd]:
```suggestion

- Change default Python version from 2.7 to 3.5
```

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 2]
```diff
@@ -181,14 +181,48 @@ set this to have values such as ``data_source`` or ``manage_data``.</xs:document
...
+
+20.05
+- json config files:
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/xsd/galaxy.xsd:212]:
```suggestion

- json config files: 
```

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 3]
```diff
@@ -181,14 +181,48 @@ set this to have values such as ``data_source`` or ``manage_data``.</xs:document
...
+- Use python 3.5 as default (before 2.7)
+
+20.05
+- json config files: 
+   - unselected optional `select` and `data_column` parameters get `None` instead of `"None"`
+   - multiple `select` and `data_column` parameters are lists (before comma separated string)
+
+20.09
+
+- use strict shell (`set -e`) for tool execution
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/xsd/galaxy.xsd]:
```suggestion
- Exit immediately if a command exits with a non-zero status (`set -e`)
```

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 4]
```diff
@@ -181,14 +181,48 @@ set this to have values such as ``data_source`` or ``manage_data``.</xs:document
...
+
+18.01
+
+- Use a separate home directory for each job.
+- Introduce `provided_metadata_style` with default `"default"` before `"legacy"`.
+
+18.09 
+
+- References to other inputs need to be fully qualified by using `|`.
+- do not allow provided but illegal default values.
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/xsd/galaxy.xsd]:
```suggestion
- Do not allow provided but illegal default values.
```

File: lib/galaxy/tool_util/xsd/galaxy.xsd [hunk 5]
```diff
@@ -181,14 +181,48 @@ set this to have values such as ``data_source`` or ``manage_data``.</xs:document
...
+
+20.05
+- json config files: 
+   - unselected optional `select` and `data_column` parameters get `None` instead of `"None"`
+   - multiple `select` and `data_column` parameters are lists (before comma separated string)
+
+20.09
+
+- use strict shell (`set -e`) for tool execution
+- assume sort order for collection elements
```
Comment (nsoranzo) [inline lib/galaxy/tool_util/xsd/galaxy.xsd]:
```suggestion
- Assume sort order for collection elements
```

**Comment by nsoranzo** [comment]
@galaxybot test this

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**